File: Leetcode/Leetcode-DSA-course/Hashing/49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> groups = new HashMap<>();
        for (String s : strs) {
            char[] arr = s.toCharArray();
            Arrays.sort(arr);
            String key = new String(arr);
            if (!groups.containsKey(key))
                groups.put(key, new ArrayList<String>());
            groups.get(key).add(s);
        }
        return new ArrayList(groups.values());
    }
}

File: Leetcode/Leetcode-DSA-course/Hashing/49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;
        for (string& s : strs) {
            string t = s;
            sort(t.begin(), t.end());
            groups[t].push_back(s);
        }
        vector<vector<string>> ans;
        for (auto [key, val] : groups) {
            ans.push_back(val);
        }
        return ans;
    }
};